types/wlr_output.c

```diff
@@ -28,8 +28,7 @@ static void wl_output_send_to_resource(struct wl_resource *resource) {
 	if (version >= WL_OUTPUT_MODE_SINCE_VERSION) {
 		struct wlr_output_mode *mode;
 		wl_list_for_each(mode, &output->modes, link) {
-			// TODO: mode->flags should just be preferred
-			uint32_t flags = mode->flags;
+			uint32_t flags = mode->flags & ~WL_OUTPUT_MODE_PREFERRED;
```

Comment:
Does this flag break things?

Reply:
I did this backwards.

Reply:
Yeah, so:
* `WL_OUTPUT_MODE_PREFERRED` has only one bit on
* `~WL_OUTPUT_MODE_PREFERRED` has only one bit off, the preferred bit (all others on)
* `mode->flags & ~WL_OUTPUT_MODE_PREFERRED` is `mode->flags` with the preferred bit off

But the comment says the contrary: `mode->flags` should only be able to change `WL_OUTPUT_MODE_PREFERRED` (ie. should not be able to set `WL_OUTPUT_MODE_CURRENT`).

Reply:
I already said I did it backwards -_-

Reply:
Oh, sorry. I though you were speaking about the `~`, saying it inverts bits so it would be fine.